freshmaker/views.py

```diff
@@ -756,9 +773,12 @@ class PullspecOverrideAPI(MethodView):
     @freshmaker_build_api_latency.time()
     def get(self, id):
         build = models.ArtifactBuild.query.filter_by(id=id).first()
-        # 'bundle_pullspec_overrides' should already contain valid JSON
         if build:
-            return jsonify(build.bundle_pullspec_overrides), 200
+            pullspec_overrides = build.bundle_pullspec_overrides
+            # remove old pullspec from output, since it used only internally for manual rebuilds
+            if pullspec_overrides and "_old" in pullspec_overrides:
```

Comment:
Isn't `pullspec_overrides` a list of dictionaries? I think you'll need to loop through each pullspec and delete the `_old` key if present.

Reply:
Yes, you are right, I will change it.

Reply:
@mprahl Wait, I just realized that pullspec_overrides is a disc that looks like:
```
{
  "update": csv_modification,
  "pullspecs": [{pullspec_1}, {pullspec_2}]
}
```
Right? So I need to check `if pullspec_overrides and "pullspecs" in pullspec_overrides:` instead. And then go through all pullspecs.

Reply:
And API endpoint should not return a list of dicts, but dict with `pullspecs` key that should contain all the pullpecs dicts, right?

Reply:
@apaplaus ah right. So you want to return `bundle_pullspec_overrides` as is except for you want to remove any `_old` key in the dictionaries in the list in `pullspecs`.

Reply:
ok, I will adjust tests then